### ISSUE TITLE
Font selection dialog is broken

### ISSUE THREAD

**Issue by Wuzzy2** [minetest/minetest#8978]
##### Minetest version
```
d77ea764768f5cde9e8f0781356a67d723aa7609
```

##### Summary
If you want to select a new font in the All Settings menu and go through the file browser menu, then select any TTF font and hit “OK”, you will see an empty gray window. I can Esc easily, but the font selection was ignored.

I also see this in console:

```
2019-09-23 14:58:11: ERROR[Main]: Invalid size element (4): '10,4,9,true'
2019-09-23 14:58:11: ERROR[Main]: Invalid pos for element boxspecified: "0,000000,0,200000"
2019-09-23 14:58:11: ERROR[Main]: Invalid pos for element textareaspecified: "0,300000,0,200000"
2019-09-23 14:58:11: ERROR[Main]: Invalid pos for element fieldspecified: "0.28,3,65"
2019-09-23 14:58:11: ERROR[Main]: Invalid pos for element buttonspecified: "8,3,35"
2019-09-23 14:58:11: ERROR[Main]: Invalid pos for element buttonspecified: "2,5,4,25"
2019-09-23 14:58:11: ERROR[Main]: Invalid pos for element buttonspecified: "5,4,25"
```

Typing in the font by hand (without opening the file browser) still works.

##### Steps to reproduce
* Go to All Settings
* Select a font (for example, in `font_path` setting)
* Go through the file browser, select TTF font and hit OK

EXPECTED: New font is selected without issues
ACTUAL: I see an empty window and font selection fails

**Comment by sfan5** [comment]
sounds like a revival of #4044

**Comment by Wuzzy2** [comment]
Only kinda sorta. It's different this time.

This time, the window that appears is completely empty. It's simply a gray formspec rectangle without anything else.

If you Esc out of it, everything is back to normal (and the font selection is discarded).
So yeah, Minetest still screws up, just not as badly as in the other issue. :D

Oh, and it's not only the font selection. I have just learned that. It's ALL settings that use the file browser.

**Comment by DS-Minetest** [comment]
Tried to reproduce with current master.
Sometimes there seems to be a segfault in irrlicht's file opener class.
And somehow sometimes the translation doesn't work correctly anymore for some things after changing the path.

**Comment by sfan5** [comment]
Can we get rid of the dialog? It'd allow deleting the code from IrrlichtMt too.

**Comment by rubenwardy** [comment]
The dialog works for me. That error is a formspec error, and the font selection dialog isn't formspecs. I suspect that the number locale was set incorrectly, and `4.9` was being rendered as `4,9`

**Comment by sfan5** [comment]
Works for me too, now.

**Comment by rubenwardy** [comment]
This was fixed by 8fe753c3d96957cb9d58d9f82677b993effe3652
